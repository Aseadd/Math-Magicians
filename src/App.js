import React from 'react';
import './App.css';
import Calculator from './components/calculator';

// eslint-disable-next-line react/prefer-stateless-function
class App extends React.Component {
  render() {
    return (
      <section>
        <h1>Home Page</h1>
        <Calculator />
      </section>
    );
  }
}

export default App;
