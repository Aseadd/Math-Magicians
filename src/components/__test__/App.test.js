import React from 'react';
import ReactDOM from 'react-dom/client';
import renderer from 'react-test-renderer';
import App from '../../App';

it('should render the app component', () => {
  const wrapper = ReactDOM.createRoot(document.createElement('div'));
  wrapper.render(<App />);
});

it('should match the snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
