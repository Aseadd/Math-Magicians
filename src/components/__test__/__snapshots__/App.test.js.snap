// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section>
  <h1>
    Home Page
  </h1>
  <section
    className="calc"
  >
    <span>
      Lets do some Math
    </span>
    <div
      className="container"
    >
      <div
        className="grid-container"
      >
        <div
          className="item1"
        >
          <p
            className="result"
          >
            0
          </p>
        </div>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="item2"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="item2"
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="item2"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="item2"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
        <button
          className="item18"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="item2"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </section>
</section>
`;
